pygen/pygen_src/test/riscv_instr_base_test.py

```diff
@@ -28,25 +30,28 @@ def __init__(self):
         self.asm_file_name = cfg.argv.asm_file_name
         self.asm = ""
 
-    def run_phase(self):
-        for _ in range(cfg.num_of_tests):
-            self.randomize_cfg()
-            self.asm = riscv_asm_program_gen()
-            riscv_instr.create_instr_list(cfg)
-            if cfg.asm_test_suffix != "":
-                self.asm_file_name = "{}.{}".format(self.asm_file_name,
-                                                    cfg.asm_test_suffix)
-            test_name = "{}_{}.S".format(self.asm_file_name,
-                                         _ + self.start_idx)
-            self.asm.get_directed_instr_stream()
-            self.asm.gen_program()
-            self.asm.gen_test_file(test_name)
+    def run_phase(self, num):
+        self.randomize_cfg()
+        self.asm = riscv_asm_program_gen()
+        riscv_instr.create_instr_list(cfg)
+        if cfg.asm_test_suffix != "":
+            self.asm_file_name = "{}.{}".format(self.asm_file_name,
+                                                cfg.asm_test_suffix)
+        test_name = "{}_{}.S".format(self.asm_file_name,
+                                     num + self.start_idx)
```

Comment:
if we pass in `num=10` here for `riscv_rand_instr_test`, won't all 10 subprocesses be overwriting in the same file?
looks like each of the 10 files will be named `riscv_rand_instr_test_10.S` ?